gimmik/__init__.py

```diff
@@ -19,11 +20,22 @@ def generate_mm(mat, dtype, platform, alpha=1.0, beta=0.0, funcn='gimmik_mm'):
     else:
         raise ValueError('Invalid floating point data type')
 
+    if 0 < (n is None) + (ldb is None) + (ldc is None) < 3:
+        raise ValueError('Must provide all of (n, ldb, ldc) or none')
+
```

Comment:
Should you just get the user to provide this as a 3-tuple?

Reply:
I considered that, but I quite like being able to pass them as named arguments `n=..., ldb=..., ldc=...` which is not possible with a tuple.  This also gives us the option of relaxing the restriction in the future should a good use case arise (currently the extra complexity in the templates is not worth it).